### PR TITLE
Feature budget app

### PR DESCRIPTION
**Pull request by Favourezeugwa**
## I implemented the following in this pull request;
- Splash screen
A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)

- When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

- The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

- Technical requirements
Used Postgres as your database.
Used devise for authentication.
Used cancancan for authorization

- The project is deployed and accessible online.
Heroku

### CODE REVIEW

File: app/views/expense_categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            <td><%= expense.name %></td>
+            <td><span>$<%= expense.amount %></span></td>
+            <td><%= link_to "Show", expense_path(expense) %>
+            <%= link_to "Delete", expense_path(expense), data: {
+                    turbo_method: :delete,
+                    turbo_confirm: "Are you sure?"
+                  } %></td>
+          </tr>
+        <% end %>
+      </tbody>
```
Comment (Deepakdanger) [inline app/views/expense_categories/index.html.erb]: - [ ] The design must be follow. Please make sure to follow the design guideline what given.


### Given design
Screenshot from 2022-10-28 05-04-17


### Your design

Screenshot from 2022-10-28 05-05-18

File: README.md [hunk 2]
```diff
@@ -1,2 +1,105 @@
...
+Feel free to check the [issues page](https://github.com/Favourezeugwa/Budget-app/issues).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to [Microverse](https://bit.ly/MicroverseTN) for giving me this project to work on
+
```
Comment (Deepakdanger) [inline README.md]: - [ ] The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: spec/features/category_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      expect(page).to have_content('Destroy')
+    end
+
+    it 'displays new category' do
+      visit new_category_path
+      expect(page).to have_content('Name')
+      expect(page).to have_content('Icon')
+      expect(page).to have_content('New category')
+      expect(page).to have_content('Back to categories')
+    end
```
Comment (Deepakdanger) [inline spec/features/category_index_spec.rb]: - [ ] Here the specs are failing. Please fix that.

Screenshot from 2022-10-28 05-16-50

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Favourezeugwa, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required changes
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/expense_categories/index.html.erb (hunk 1 above)
Comment (Favourezeugwa) [reply app/views/expense_categories/index.html.erb]: Hello @Deepakdanger , please I have this page on my categories. The page that you have screenshotted is showing all transactions in each category. Please kindly check my categories page, and you will find what you are looking for. As you can see, there are different icons showing in each category just like the template.

By the way, the Behance design that was provided by microverse was only one-page styling. There are no proper designs for how all pages should look like. I have implemented the style provided by microverse on my CATEGORIES page.

Thanks.

File: README.md (hunk 2 above)
Comment (Favourezeugwa) [reply README.md]: @Deepakdanger I will fix this

File: spec/features/category_index_spec.rb (hunk 3 above)
Comment (Favourezeugwa) [reply spec/features/category_index_spec.rb]: Will fix this too

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:name, :email, :password, :password_confirmation)
+    end
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :email, :password, :current_password, :password_confirmation)
+    end
+  end
```
Comment (youmari) [inline app/controllers/application_controller.rb:14]: - [ ] [OPTIONAL] I wasn't able to sign up through sign up page please make sure to fix that in order your app will be working properly

**Review by youmari — APPROVED** [review]
Hi Friend @Favourezeugwa you are doing Great  , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (Favourezeugwa) [reply app/controllers/application_controller.rb:14]: Hi @youmari , thank you for your review. I will fix this. 🤗 🤗

**Pull request merged**